### ISSUE TITLE
Some package ID characters are rejected by IIS request filtering

### ISSUE THREAD

**Issue by joelverhagen** [NuGet/NuGetGallery#6817]
I encountered this on dev when playing with a zalgo package ID.

## Steps

1. Start a UI upload of a package that has zalgo text in the package ID.
1. Upload the package

## Expected

The user should be redirected to the package details page with a message like:

```
You successfully uploaded {ID} {version}.
```

## Actual

HTTP 400 IIS error page.

## Cause

I think this is because the server sends back the following cookie after upload, but before landing on the display package page:

```
Cookie name:    __Controller::TempData
Cookie value:   Message=You successfully uploaded z̡̜͍̈̍̐̃̊͋́a̜̣͍̬̞̝͉̽ͧ͗l̸̖͕̤̠̹̘͖̃̌ͤg͓̝͓̰̀ͪo͈͌ 1.0.0.
```

The client browse accepts this cookie then sends it back to the display package page. This gets rejected by IIS, thus resulting in IIS 400.

![image](https://user-images.githubusercontent.com/94054/51067675-4282f800-15c9-11e9-90d6-5fd08ff4e53e.png)

**Comment by scottbommarito** [comment]
IMO we probably shouldn't allow "zalgo" characters in package IDs.
We definitely shouldn't, however, say that a package was uploaded when it actually was rejected.

**Comment by joelverhagen** [comment]
The package was accepted by gallery, but rejected by our validation pipeline (due to a potential bug).

**Comment by skofman1** [comment]
Do you think it's important for us to support zalgo characters?

**Comment by joelverhagen** [comment]
Not necessarily. There are two angles that should be explored:

1. Are any characters in use today on nuget.org rejected for this reason? This basically makes nuget.org unusable until the user thinks to clear his cookies. Every page returns that 400.

1. (as you alluded) Can we restrict the characters like zalgo since it's pretty ridiculous to call your package this? ASCII is too narrow but maybe there is a sweet spot that is still more restrictive than what we have today, which is `\w` in .NET's regex implementation.

**Comment by skofman1** [comment]
> Can we restrict the characters like zalgo since it's pretty ridiculous to call your package this? ASCII is too narrow but maybe there is a sweet spot that is still more restrictive than what we have today, which is \w in .NET's regex implementation.

Agree. Let's add this to backlog. We will figure out the exact character set when it becomes a problem.
